### DISCUSSION TITLE
Search for assets should show path to asset

### DISCUSSION THREAD

**Discussion by bossanova808** [craftcms/cms#11450] [Ideas] [answered]
For those of us with a _lot_ of assets, created by many people over many years, it would be super handy if, when searching for an asset (with 'include subfolders' ticked, which should be the default really...), it showed you what path the matches were actually found in...and ideally, there would be an option to easily navigate to that path, too.

(I know you can click on an asset and see the path there, but then if you go back to the main view, the search term is lost and that's an irritant).

An extra column here with 'path' would be awesome...

![Where Are They Eh](https://user-images.githubusercontent.com/731309/173960677-4f8bca68-5d29-4d31-8a91-3767a8a57f38.png)

**Comment by brandonkelly** [comment, marked as answer]
I’ve just added a new “Location” table column option to assets, for Craft 4.1. (0f9ac1b0c0e4e1843370ac3fb2b3fe6a8411b5da)

It won’t get added automatically when searching, but you can include it in the table ahead of time, for volumes where it’s likely to be needed.

<img width="675" alt="An asset index table with a “Location” column" src="https://user-images.githubusercontent.com/47792/174447175-97432162-904c-40eb-b0f4-c7595f6bda24.png">

Assets’ “Location” meta values have also been improved a bit to be consistent with Location table cells, so each segment of the path now links off to the referenced volume/folder.

<img width="313" alt="An asset’s metadata table, showing that the “Location” value contains hyperlinks to the volume and subfolder the asset lives in" src="https://user-images.githubusercontent.com/47792/174447165-64ee576a-c1d7-4011-9e3c-9855a7b264a5.png">

**Reply by bossanova808** [reply]
Excellent - thanks Brandon!  Another reason for me to get a move on with my update to Craft 4 work...

**Reply by brandonkelly** [reply]
Craft 4.1 is out with those changes ✨

**Reply by bencresty** [reply]
You guys are on fire! So quick
